src/redis.c

```diff
@@ -682,6 +682,10 @@ static void redis_read_server_info(redis_node_t *rn) {
                     "total_net_input_bytes", DS_TYPE_DERIVE);
   redis_handle_info(rn->name, rr->str, "total_bytes", "output",
                     "total_net_output_bytes", DS_TYPE_DERIVE);
+  redis_handle_info(rn->name, rr->str, "current", "master_repl_offset",
```

Comment:
What do you think about something like this: "current_repl_offset" as type and "master" as type_instance? In my opinion "current" as a type can be too generic.

Reply:
i`m select current because this type already exists in types.db :)

Reply:
`current` in types.db is in terms of electrical engineering meaning, so please choose another type.

Reply:
maybe "flow" or "gauge" or "records" ?

Reply:
No flow, no gauge.
These offsets are in records or bytes?

I think you might add new type `offset` and set type instance to `master_repl` or `slave_repl`.

BUT:

Another question: why do you need these metrics? 
From my POV, they are not informative, their value grow monotonically and can't be placed on charts.

Just looked into my servers, results are:

```
role:slave
slave_repl_offset:58271319441
master_repl_offset:58271319441
```

```
role:master
connected_slaves:1
...
master_repl_offset:58272087142
```

So I think these raw values are not useful in wild.
Please argue, why they are needed in form you try to propose.

Reply:
Offset measured in records.
I didn’t want to add a new type in types.db as this will increase the file size and give absolutely nothing, the name of the metric fully reflects the essence.
My case:
Compare two or more offsets, between master and slave/slaves per fixed interval,
the result will be lagg master-slave(s) in ops(records)
If lagg > x or slaves < n - send alarm message.

Reply:
> Can you please provide charts example built based on these metrics?

stack: grafana<-graphite<-collectd
https://notify.ekb-info.ru/uploads/2019-11-22-13-52-37-tsqnalxdapbenddx.png
https://notify.ekb-info.ru/uploads/2019-11-22-13-55-40-fwtmzkbdncmijcnf.png

Reply:
>Compare two or more offsets, between master and slave/slaves per fixed interval,
the result will be lagg master-slave(s) in ops(records)

Then this needs to be done on Collectd level, so its `threshold` plugin could be used to sent notifications.

Collectd is not agent used only for graphana.
Personally, I dislike implementations when Collectd is used to sent raw metrics which do not represent nothing by itself, for future calculations which could be done on Collector side too.